Architecture.md

```diff
@@ -182,10 +189,65 @@ As described previously, the API manages most of the user interactions and redir
 
 ### Database architecture
 
-**_TODO: [CLIENT] describe the database architecture, as referenced in the MCD in the README_**
+As the database is unified and serves as a focal point of the application, everything is contained in the same logical database.
+The architecture is as follows:
+
+```mermaid
+erDiagram
+    USERS {
+        uuid id PK
+        varchar(40) username UK
+        varchar(100) email UK
+        varchar(255) password
+        bool is_admin "Default false"
+        uuid email_nonce UK "Nullable, default uuid()"
+        uuid password_nonce UK "Nullable"
+        timestamp created_at
+        timestamp updated_at "Nullable"
+    }
+    SSH_KEYS {
+        uuid id PK
+        text value "public SSH key content"
+        varchar(30) name
+        timestamp created_at
+        timestamp updated_at "Nullable"
+        uuid user_id FK
+    }
+    PROJECTS {
+        uuid id PK
+        varchar(40) name
+        jsonb config
+        timestamp created_at
+        timestamp updated_at "Nullable"
+        uuid user_id FK
+    }
+
+    USERS ||--o{ PROJECTS : manage
+    USERS ||--o{ SSH_KEYS : possess
+```
+
+As you can notice, every table has a UUID field as a primary key. Compared to [Serial](https://www.postgresql.org/docs/current/datatype-numeric.html) that is often used to identify a table, a UUID guarantees a better uniqueness across the whole database. 
+To generate a [UUID](https://www.postgresql.org/docs/current/datatype-uuid.html), we are using the UUIDv4 standard, which generates each UUID randomly. Thus, there are roughly 103 trillion UUIDv4s, lowering the chance of finding a duplicate UUID to one-in-a-billion. 
+In comparison, Serials are limited to roughly 2 billion (2147483647) and will eventually clog up the column and be reduntant. As the column is a primary key, this results in the table being full and creates an inability to insert new rows. 
+However, even though serials take up less space (4 bytes) than a UUID (16 bytes), the column becomes a series of incrementing integers and offers information about the time of creation. This makes the table prone to timestamp-guessing. UUIDs are generated randomly and impossible to guess, completely removing this attack pattern.
+
+
+Furthermore, since the projects will be deployed at a subdomain that is named after `{projectId}.user-app.paastech.cloud` we needed to hide the internal database structure that emerges with a serial id. Therefore, using a uuid prevents sharing sensitive data with everyone and prevents targeted attacks.
```

Comment:
```suggestion
Furthermore, since the projects will be deployed at a subdomain that is named after `{projectId}.user-app.paastech.cloud` we needed to hide the internal database structure that emerges with a serial id. Therefore, using a UUID prevents sharing sensitive data and targeted attacks.
```